src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -232,6 +530,51 @@ IMF_STD_ATTRIBUTE_DEF (timeCode, TimeCode, TimeCode)
 
 IMF_STD_ATTRIBUTE_DEF (framesPerSecond, FramesPerSecond, Rational)
 
+//
+// timecodeRate -- the nominal playback rate associated with
+// the image sequence to which the image belongs, specified in
+// timecodes per second.
+//
+// The integer timecodeRate attribute is a rounding of the
+// framesPerSecond attribute.
+//
+// The rounding is atypical: in the special case of the nominal
+// playback rate corresponding exactly to the midpoint of two
+// adjacent integer values, timecodeRate is defined to always
+// be the rounding up to the next larlgest integer.
+//
+// This stands in contrast to the default IEEE 754 rounding for
+// decimal arithmetic in which such a midpoint value would always
+// be rounded to the nearest even integer.
+//
+
+IMF_STD_ATTRIBUTE_DEF (timecodeRate, TimecodeRate, int)
```

Comment:
Why would we introduce this (as opposed to telling people to use a rational rate with a divisor of 1.0)? It's another opportunity for the file to disagree with itself if both attributes are present.

Assuming there's a "rational" (haha) reason to have this attribute, I think the comment about atypical rounding should indicate that since that NTSC rates mangle rates with a 1.001 divisor, the atypical rounding is meant as a cruel hack for snapping to an integer value. At least my supposition is that that's the motivation for a snap-up.

Reply:
timecodeRate could still be useful for High Framerate (HFR) sequences. The timecode 'frames' field only goes up to 29, so there's an implicit 30fps maximum rate. Also, there's usually a need for a standard framerate version of the sequence. Having two different definitions of timecode would create confusion on a project, so it makes sense always to have timecode refer to the standard framerate version. In that case, you might have frameRate at 120fps and timecodeRate at either 30fps or 24fps, in which case either 4 or 5 sequential frames would have the same timecode.

Reply:
wrt the latest patch removing the field, high framerate seems like a topic to come back to down the line.